### ISSUE TITLE
gitignore not respected in certain situations?

### ISSUE THREAD

**Issue by joshaber** [libgit2/libgit2#3143]
1. `mkdir test; cd test; git init`
2. Use this `.gitignore`:

```
*.com

tags
!tags/

.buildpath
```
3. `touch .buildpath`
4. `git status`. Note that `.buildpath` is properly ignored by git.git.
5. Get status using libgit2. Note that `.buildpath` is not ignored.

**Comment by joshaber** [comment]
(This is on 63e8b08dfcaa039e7bbbcfa4b5cdcea8bddf13c0)

**Comment by carlosmn** [comment]
I'mma blame @ethomson as he's the one who touched ignores last :p

It's reproducible with a simple test, though it seems to need this sequence of rules, removing any of them makes the test pass.

```C
void test_status_ignore__ignored_ignore(void)
{
	int ignored;

	g_repo = cl_git_sandbox_init("empty_standard_repo");

	cl_git_mkfile("empty_standard_repo/.gitignore", "*.com\ntags\n!tags/\n.buildpath");
	cl_git_mkfile("empty_standard_repo/.buildpath", "");
	cl_git_pass(git_ignore_path_is_ignored(&ignored, g_repo, ".buildpath"));
	cl_assert_equal_i(1, ignored);
}
```

**Comment by ethomson** [comment]
Dang.  I was hoping that it had been fixed since 63e8b08 by accident, since we did some refactoring in there.

Despite that, you can still point the finger at me.  ;)

**Comment by Therzok** [comment]
![](http://media.giphy.com/media/OMKZfNdSPpXfa/giphy.gif)
